### PR TITLE
Feature/add page timings state

### PR DESCRIPTION
**Pull request by TumkoVolodymyr**

### CODE REVIEW

File: frontend/src/pages/PageTimings.vue [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+                chartData: GET_LINE_CHART_DATA,
+            }),
+            pieData () {
+                return [
+                    ['Type', 'Value'],
+                    ['New Visitors', this.pieChartData.newVisitors],
+                    ['Return Visitors',this.pieChartData.returnVisitors]
+                ];
+            },
+            legend () {
```
Comment (wyyi) [inline frontend/src/pages/PageTimings.vue]: it seems this property is not used anywhere

File: frontend/src/pages/PageTimings.vue [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            };
+        },
+        computed: {
+            ...mapGetters('page_timings', {
+                buttonsData: GET_BUTTON_DATA,
+                currentActiveButton: GET_ACTIVE_BUTTON,
+                getSelectedPeriod: GET_SELECTED_PERIOD,
+                chartData: GET_LINE_CHART_DATA,
+            }),
+            pieData () {
```
Comment (wyyi) [inline frontend/src/pages/PageTimings.vue]: we don't seem to have a pie chart on this page, do we?

File: frontend/src/pages/PageTimings.vue [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+                };
+            },
+        },
+        created () {
+            this.fetchPageData();
+        },
+        methods: {
+            ...mapActions('page_timings', {
+                changeActiveButton: CHANGE_ACTIVE_BUTTON,
+                changeFetchingButtonState: CHANGE_FETCHED_BUTTON_STATE,
```
Comment (wyyi) [inline frontend/src/pages/PageTimings.vue]: this action seems to be unsed

File: frontend/src/store/modules/page_timings/actions.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        context.commit(SET_SELECTED_PERIOD, payload.value);
+        context.dispatch(FETCH_PAGE_DATA);
+    },
+    [CHANGE_ACTIVE_BUTTON]: (context, button) => {
+        context.commit(SET_ACTIVE_BUTTON, button);
+        context.dispatch(FETCH_LINE_CHART_DATA);
+        context.dispatch(FETCH_TABLE_DATA);
+
+    },
+    [CHANGE_FETCHED_BUTTON_STATE]: (context, data) => {
```
Comment (wyyi) [inline frontend/src/store/modules/page_timings/actions.js]: i think we could do without this one

File: frontend/src/store/modules/page_timings/actions.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+            .then(response => {
+                context.commit(SET_BUTTON_DATA, {button: type, value: response.value});
+                context.commit(RESET_BUTTON_FETCHING, type);
+            })
+            .catch(err => {
+                context.commit(RESET_BUTTON_FETCHING, type);
+                throw err;
+            });
+    },
+    [CHANGE_FETCHED_LINE_CHART_STATE]: (context, value) => {
```
Comment (wyyi) [inline frontend/src/store/modules/page_timings/actions.js]: and this one

File: frontend/src/store/modules/page_timings/actions.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+            .fetchChartValues(startDate.unix(), endDate.unix(), period.interval)
+            .then(data => context.commit(SET_LINE_CHART_DATA, data))
+            .finally(() => context.commit(RESET_LINE_CHART_FETCHING));
+
+    },
+    [CHANGE_GROUPED_PARAMETER]: (context, parameter) => {
+        context.commit(SET_GROUPED_PARAMETER, parameter);
+        context.dispatch(FETCH_TABLE_DATA);
+    },
+    [CHANGE_FETCHED_TABLE_STATE]: (context, value) => {
```
Comment (wyyi) [inline frontend/src/store/modules/page_timings/actions.js]: and probably this one too

Comment (TumkoVolodymyr) [reply frontend/src/store/modules/page_timings/actions.js]: in task for linking particular component can delete unused methods

**Pull request merged**